client/src/components/student/RatingModal.jsx

```diff
@@ -0,0 +1,112 @@
+    Practical: '',
+    WorkEthic: ''
+  });
+
+  const handleCommentChange = (field, value) => {
+    setComments((prevComments) => ({
+      ...prevComments,
+      [field]: value
+    }));
+  };
+
+  const onSubmit = async (e) => {
+    e.preventDefault();
+    if (!window.confirm("Are you sure you want to submit?")) return;
+
+    
+    await request("/courses/add-review", "POST", {
+      reviewer_id : user.user_id,
+      reviewee_id : revieweeId,
+      group_id : groupId,
+      review: {
+        cooperation: ratingCooperation,
+        conceptual: ratingConceptual,
+        practical: ratingPractical,
+        work_ethic: ratingWorkEthic,
+        cooperation_comment: comments.Cooperation,
+        conceptual_comment: comments.Conceptual,
+        practical_comment: comments.Practical,
+        work_ethic_comment: comments.WorkEthic
+      }
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error("Couldn't add review");
+        }
+        return res;
+      })
+      .then(res => handleClose())
+      .catch((e) => {
+        console.error(e);
+      });
+  };
+
+  return (
+    <div className="flex align-center justify-center text-white">
+      <div className="bg-black w-fit text-white flex flex-col gap-4 p-40 rounded-xl">
+        <h2 className="text-center text-white text-3xl">Enroll in a course</h2>
+        <form onSubmit={onSubmit} className="flex flex-col gap-3">
+          <div className="grid grid-cols-2 gap-4">
+            {['Cooperation', 'Conceptual', 'Practical', 'Work Ethic'].map((field, idx) => (
+              <div key={idx} className="flex flex-col items-center p-4 border border-white rounded-lg">
+                <label className="mb-2">{field}</label>
+                <Rating
+                  onClick={(rate) => criteria[idx](rate)}
+                />
+                <textarea
+                  className="text-black mt-2 w-full"
+                  placeholder={`${field} Comment`}
+                  value={comments[field]}
+                  onChange={(e) => handleCommentChange(field.replace(/\s+/g, ''), e.target.value)}
+                />
+              </div>
+            ))}
+          </div>
+          <input
+            type="submit"
+            className="text-m w-fit self-center text-center border-solid border-white border-[3px] ease-in duration-50 p-1 rounded-lg hover:scale-110"
+          />
+        </form>
+        <button
+          onClick={(e) => {
+            e.preventDefault();
+            handleClose();
+          }}
+          className="mt-4"
+        >
+          Cancel
+        </button>
+      </div>
+    </div>
+  );
+};
+
+export default RatingModal;
+
+
+{/* <button
+  className="text-xl border-solid border-2 w-fit p-2 rounded-md self-center"
+  onClick={() => setIsReviewing(true)}
+>
+  Add Review
+</button>
+<Modal open={isReviewing} onClose={() => setIsReviewing(false)}>
+  <RatingModal
+    handleClose={() => setIsReviewing(false)}
+  />
+</Modal> */}
```

Comment:
Is this for when @Kerryki stuff is ready?

Reply:
Yup it needs to be added on a user in a group